### PR TITLE
Explicit type annotations and !isEmpty replace with nonEmpty

### PR DESCRIPTION
**Pull request by sh0hei**

### CODE REVIEW

**Review by SethTisue — APPROVED** [review]

**Comment by adriaanm** [comment]
I think this is good for inclusion in 2.12.6

File: src/compiler/scala/tools/nsc/ast/parser/xml/Utility.scala [hunk 1]
```diff
@@ -77,7 +77,7 @@ object Utility {
       else sb append c
     }
 
-    if(!sb.isEmpty) // flush buffer
+    if(sb.nonEmpty) // flush buffer
```
Comment (som-snytt) [inline src/compiler/scala/tools/nsc/ast/parser/xml/Utility.scala:80]: oh man, missed opportunity to normalize spacing before the LPAREN. I promise to contribute more reviews on lunch breaks. Or maybe on snack breaks. Also, in a demoralizing blow, retronym recently came out against `nonEmpty`, in some context. Or is this scala StringBuilder with native nonEmpty, and why do I have to ask?

Comment (adriaanm) [reply src/compiler/scala/tools/nsc/ast/parser/xml/Utility.scala:80]: Good question -- I didn't think to check, but we get `TraversableOnce's` version here: `def nonEmpty: Boolean = !isEmpty` (according to IntelliJ). So, neither `isEmpty` nor `nonEmpty` uses the potentially more efficient `length` on the underlying Java `StringBuilder`.

**Pull request merged**